entrypoint.sh

```diff
@@ -28,8 +24,8 @@ then
 fi
 
 export RAILS_RELATIVE_URL_ROOT=${APPLICATION_ROOT:-'/app/root'}
-export SCRIPT_NAME=${APPLICATION_ROOT}
+export SCRIPT_NAME=${RAILS_RELATIVE_URL_ROOT}
```

Comment:
What is the point of setting `SCRIPT_NAME` via this environment variable?
This can only break the system.

Reply:
@andrew-pickin-epi `SCRIPT_NAME` is used to "[... ] have both the assets paths and path helpers configured for serving a Rails application from a subdirectory." [^1]

It is similar to `RAILS_RELATIVE_URL_ROOT` in that "[...] `SCRIPT_NAME` isn't available during the asset compilation task so config.relative_url_root needs to be set so that it knows where they will be served from by the webserver. This is especially important for Sass since the asset helpers like image-url, etc. will generate absolute urls." [^2]

I don't follow how this can break the system though as these are build time env vars, NOT runtime env vars and we went through all this together?

[^1]:https://jlintusaari.net/deploying-rails-application-to-subdirectory/
[^2]:https://github.com/rails/rails/issues/8119#issuecomment-25442568

Reply:
You can break the system is `RAILS_RELATIVE_URL_ROOT` different at run-time from what it's set to at build-time (or if either of these is different from `SCRIPT_NAME`).